### ISSUE TITLE
Oracle - unable to insert rows into table without primary key?

### ISSUE THREAD

**Issue by RustyF** [fsprojects/SQLProvider#514]
I'm currently using the Oracle Provider to insert rows but hitting a KeyNotFoundException during ctx.SubmitUpdates().  I think is due to the table not having any primary key. I think this is the match expression in the ProcessUpdates() code that fails: -

````
match e.GetPkColumnOption primaryKeyColumn.[e.Table.Name].Column with
| [] ->  e.SetPkColumnSilent(primaryKeyColumn.[e.Table.Name].Column, id)
| _ -> () // if the primary key exists, do nothing
          // this is because non-identity columns will have been set
          // manually and in that case scope_identity would bring back 0 "" or whatever
````
i.e. there are no entries in the primary key dictionary keyed on the table name.  

Just wondering if this is intentional or whether we need to account for this case?

**Comment by Thorium** [comment]
At least the row cannot be updated. Maybe insert could be allowed still? So the code change would be just something like

```fsharp
  let! id = cmd.ExecuteScalarAsync() |> Async.AwaitTask
  if primaryKeyColumn.ContainsKey(e.Table.Name) then
      match e.GetPkColumnOption primaryKeyColumn.[e.Table.Name].Column with
      | [] ->  e.SetPkColumnSilent(primaryKeyColumn.[e.Table.Name].Column, id)
      | _ -> () // if the primary key exists, do nothing
                // this is because non-identity columns will have been set
                // manually and in that case scope_identity would bring back 0 "" or whatever
  e._State <- Unchanged
```

**Comment by Thorium** [comment]
Indeed, the other providers use `CommonTasks.checkKey pkLookup id e`

**Comment by Thorium** [comment]
Fixed in 1.1.31

**Comment by RustyF** [comment]
Fab, many thanks
